lib/components/primitive-components/Trace/create-downward-net-label-symbol.ts

```diff
@@ -0,0 +1,165 @@
+import type { SoupUtilObjects } from "@tscircuit/soup-util"
+import type { Port } from "../Port"
+
+interface CreateDownwardNetLabelSymbolParams {
+  fromPort: Port
+  toPort: Port
+  fromAnchorPos: { x: number; y: number }
+  toAnchorPos: { x: number; y: number }
+  schDisplayLabel: string
+  source_trace_id: string
+}
+
+export const createDownwardNetLabelSymbol = (
+  {
+    fromPort,
+    toPort,
+    fromAnchorPos,
+    toAnchorPos,
+    schDisplayLabel,
+    source_trace_id,
+  }: CreateDownwardNetLabelSymbolParams,
+  db: SoupUtilObjects,
+): void => {
```

Comment:
ok so I'm able to read this a bit more specifically now,

it seems like you've made this code specific to "ground" so we should actually rename the function for "ground" i.e. `createDownwardGroundSymbol`

There are many many "magic numbers" here, these should _all_ be derived from the symbol data from `schematic symbols` so that this method can be generalized to any symbol and generally it's just best to avoid any magic numbers, every number should be "derived from" something

The ternaries in this function are very hard to read, the function also has duplicated logic for both the `from` and the `to`, each of these parts should be broken out to avoid repeating essentially the same logic.

In general, you shouldn't have nested ternaries. You can try the following pattern to simplify these and make them more readable:

```ts
const schBoxOffset = {
  "left": { x: -0.5, y: 0 },
  "top": { x: 0, y: 0.5 },
  "right": { /* ... */ },
  "bottom": {/* ... */}
}[port.facingDirection]
```

etc.

Reply:
@Abse2001 you can still use the magic numbers if you rename the function to be specific to ground! I think I typo'd a bit- only the "general purpose downward symbol function" shouldn't have magic numbers